static/scripts/assignment_table.js

```diff
@@ -0,0 +1,482 @@
+
+                            //Verify if the row value for the specific column is in the filter
+                            for (const row of tab) {
+                                if (values.includes(row[col])) {
+                                    //Push id to the filteredResults array
+                                    filteredResults.push(row[0]);
+                                }
+                            }
+                        }
+                    }
+                    filtersPlugin.clearConditions();
+                    //Create a new condition to filter the data based on the id
+                    filtersPlugin.addCondition(0, 'by_value', [filteredResults]);
+                }
+            },
+            afterFilter(conditionsStack) {
+                const filtersPlugin = this.getPlugin('filters');
+                const filtersRowsMap = filtersPlugin.filtersRowsMap;
+
+                // Exclude fixed lines from the filter
+                for (let i = 0; i < lenFixedRowsText; i++) {
+                    filtersRowsMap.setValueAtIndex(i, false);
+                }
+            },
+            licenseKey: "non-commercial-and-evaluation",
+        });
+
+        $(document).ready(function () {
+            // Disable the cells that contain user preferences
+            table.updateSettings({
+                cells: function (row, col) {
+                    let cellProperties = {};
+                    // Row % 2 === 0 to get the user preferences
+                    if (row >= lenFixedRowsText && col >= lenFixedHeaders && row % 2 === 0) {
+                        cellProperties.readOnly = true;
+                    }
+                    return cellProperties;
+                }
+            });
+            function updateToastContent(message) {
+                let toastBody = document.querySelector('#toast-notification .toast-body');
+                toastBody.textContent = message;
+            }
+
+            let toastNotification = new bootstrap.Toast(document.getElementById('toast-notification'));
+
+            $('#button-export').click(function () {
+                const exportPlugin = table.getPlugin('exportFile');
+                exportPlugin.downloadFile('csv', {
+                    bom: false,
+                    columnDelimiter: ',',
+                    columnHeaders: true,
+                    exportHiddenColumns: false,
+                    exportHiddenRows: false,
+                    fileExtension: 'csv',
+                    filename: 'ICTM-CSV-file_[YYYY]-[MM]-[DD]',
+                    mimeType: 'text/csv',
+                    rowDelimiter: '\r\n',
+                    rowHeaders: true
+                });
+                updateToastContent('Data exported to CSV');
+                toastNotification.show();
+            })
+            $('#button-create-assignments').click(function () {
+                storeDataLocally(data);
```

Comment:
Do we actually want this ? Storing this in the local storage only allows the local browser to load saved data. This means:

- an admin currently working on assignments should keep using the same browser until assignments are published
- if an admin save their own assignments, another admin cannot get the current state of assignments and go on until the data are ready for publishment.

Data should be saved in remote database. And can only be viewed by teachers and researchers once they are published.